### DISCUSSION TITLE
Wrapping text in list items ?

### DISCUSSION THREAD

**Discussion by AYehia0** [charmbracelet/bubbles#329] [Q&A]
I have a simple bubbletea program where I split the screen into 2 to show 2 lists. 
I have a problem is that if the item's width/length is large > 50, only the first 50 char are displayed, rest with ...

[![2023-02-07-16-27.png](https://i.postimg.cc/dVZ6N0kD/2023-02-07-16-27.png)](https://postimg.cc/4YZpms1T)

Here is what I am doing : 
```golang
type item struct {
	title, desc  string
	ayah         quran.Ayah
	surahsInPage string
	ayahNumber   int
	endOfSurah   bool
}

func (i item) Title() string {
	ayahWidth := 80
	text := lipgloss.NewStyle().Width(ayahWidth).Render(fmt.Sprintf("%s", i.ayah.Text))
	return text
}
```

**Comment by bashbunni** [comment]
Hmmmm I know that reflow does a great job with wrapping text, but I haven't tried using it for list item titles... 
https://github.com/muesli/reflow

I'll take a quick look

**Comment by bashbunni** [comment]
Hey @AYehia0, it would work most reliably if you build your own list component. I tested out a list example with custom height and width fields, but it wouldn't render reliably at different widths. 

This example would be a great starting point for a multi-line title solution: https://github.com/charmbracelet/bubbletea/tree/master/examples/list-simple

For your reference, here's roughly what it looked like when I was making edits to the list item widths and heights using the `DefaultDelegate`:
```go
...
	delegate := list.NewDefaultDelegate()
	delegate.Styles.NormalTitle.Width(80)
	delegate.Styles.SelectedTitle.Width(80)
	delegate.Styles.NormalTitle.MaxHeight(2)
	delegate.Styles.SelectedTitle.MaxHeight(2)
	delegate.SetHeight(3)
	m := model{list: list.New(items, delegate, 0, 0)}
...
```

I hope that helps! Don't hesitate to reach out if you have any questions

**Comment by AYehia0** [comment]
Hey @bashbunni , thanks for replying.

I tried the solution you proposed, but unfortunately it doesn't display the full text as shown below : 
[![2023-02-08-07-20.png](https://i.postimg.cc/7h2fy0wC/2023-02-08-07-20.png)](https://postimg.cc/T5TRVyyG)


here is my snippet : 
```golang
...
	listDelegate.Styles.SelectedTitle.Width(40)
	listDelegate.Styles.SelectedTitle.MaxWidth(40)
	listDelegate.Styles.NormalTitle.Width(50)
	listDelegate.Styles.NormalTitle.MaxWidth(150)

	listDelegate.Styles.SelectedTitle.MaxHeight(2)
	listDelegate.Styles.NormalTitle.MaxHeight(2)
	listDelegate.SetHeight(3)
	listDelegate.SetSpacing(3)
...
```
btw, I really like your content, keep it up :smile:

**Comment by bashbunni** [comment]
@AYehia0 Thank you! I would suggest making a custom list component then. There's a lot happening under the hood with the list bubble that makes it require a fixed height from all elements in the list. That example I linked previously should be a good place to start! Let us know if you have any questions along the way

**Comment by AYehia0** [comment]
Thanks @bashbunni, quick question what do you mean by a custom list component? do you mean custom rendering method like this [here](https://github.com/charmbracelet/bubbletea/blob/331a63bdca9b7815a044294d9ceb15cb6363889e/examples/list-simple/main.go#L33).
